leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/servlet/json/SecuritySerializer.java

```diff
@@ -42,7 +42,9 @@ public JsonElement serialize(SecurityInfo src, Type typeOfSrc, JsonSerialization
         if (src.getIdentity() != null) {
             JsonObject psk = new JsonObject();
             psk.addProperty("identity", src.getIdentity());
-            psk.addProperty("key", Hex.encodeHexString(src.getPreSharedKey()));
+            if (src.getPreSharedKey() != null) {
+                psk.addProperty("key", Hex.encodeHexString(src.getPreSharedKey()));
+            }
```

Comment:
This is out of scope of this PR. Please open another PR or issue if you think this is a bug.
But to be honest, I'm not sure we want to do that :thinking: